src/recipes/tooling-editor-plugin.md

```diff
@@ -0,0 +1,56 @@
+<!--
+  ~ Copyright (c) godot-rust; Bromeon and contributors.
+  ~ This Source Code Form is subject to the terms of the Mozilla Public
+  ~ License, v. 2.0. If a copy of the MPL was not distributed with this
+  ~ file, You can obtain one at https://mozilla.org/MPL/2.0/.
+-->
+
+# Editor Plugin
+
+Using `EditorPlugin` types is very similar to the process used when
+[writing plugins in GDScript](https://docs.godotengine.org/en/stable/tutorials/plugins/editor/making_plugins.html). However,
+like Custom Classes, they
+are registered automatically by godot-rust and cannot be enabled/disabled in the Project Settings plugins pane.
+
+Plugins written in GDScript are automatically disabled if they have a code error, but because Rust is a compiled language,
+you cannot introduce compile-time errors because the library binary will simply not be emitted on the compiler finding an error.
+This is not the case for run-time errors.
+
+
+## Creating an EditorPlugin
+
+```rs
+#[derive(GodotClass)]
+#[class(tool, editor_plugin, base=EditorPlugin)]
+struct MyEditorPlugin {
+    #[base]
+    base: Base<EditorPlugin>,
+}
+
+impl IEditorPlugin for MyEditorPlugin {
+    fn init(base: Base<EditorPlugin>) -> Self {
+        // this is a prime candidate to be replaced by "init"
+        Self { base }
+    }
+
+    fn enter_tree(&mut self) {
+        // perform typical plugin operations here
+    }
+
+    fn exit_tree(&mut self) {
+        // perform typical plugin operations here
+    }
+}
+```
+
+```admonish hint
+Since this is a plugin, that custom node of yours will be added to the scene tree and visible during runtime. This means you
+can access the scene 
+tree with this. Additionally, it is safe to access the `EditorInterface` singleton through this node, which allows adding different GUI elements to 
+the editor directly. This can be incredibly helpful if you have an advanced GUI you want to implement.
+
+Use an `is_editor_hint`
+[guard](https://godot-rust.github.io/docs/gdext/master/godot/engine/struct.Engine.html#method.is_editor_hint) if 
+you don't want some code executing during runtime of the game. 
+[Read more information on guard clauses in computer science.](https://en.wikipedia.org/wiki/Guard_(computer_science)s)
+```
```

Comment:
> Since this is a plugin, that custom node of yours...
You are referring to two things, but there is only one.

I'd rather mention that:
- `EditorPlugin` _is_ a node.
- Editor plugins are added to the scene tree -- how? 
   - Manually? 
   - If automatically, where in the tree are they inserted?

---

> This can be incredibly helpful if you have an advanced GUI you want to implement.

"This can be helpful..." is also enough 🙂 

---

> ``` Use an `is_editor_hint` [guard](...) ```

The link should encompass the symbol as well:
``` [`is_editor_hint` guard](...) ```